### PR TITLE
chore: Fix dev bundlers via metro config

### PR DESCRIPTION
**Pull request by tuncaulubilge**
This PR fixes android:app ios:app and storybook-native bundlers so they pick up dev changes instead of transpiled ones.

### CODE REVIEW

File: android-app/package.json [hunk 1]
```diff
@@ -8,7 +8,7 @@
     "prettier:diff": "prettier --list-different '**/*.*'",
     "publish-library": "./publish.sh",
     "lint": "eslint . && yarn prettier:diff",
-    "start": "react-native start"
+    "start": "react-native start --config ../../../../android-app/metro.config.js"
```
Comment (nathggns) [inline android-app/package.json]: this path is wild

Comment (tuncaulubilge) [reply android-app/package.json]: They fixed this in the cli in RN 59, but for now, it's relative to node_modules/react-native/blabla.../cli

File: android-app/metro.config.js [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+    },
+    serializer: {
+      getModulesRunBeforeMainModule: () => [
+        path.join(root, 'node_modules/react-native/Libraries/Core/InitializeCore.js')
+      ],
+      getPolyfills: require(path.join(root, 'node_modules/react-native/rn-get-polyfills')),
+    },
+    server: {},
+    projectRoot: root,
+    watchFolders: [root, path.join(root, 'node_modules'), path.join(__dirname, 'node_modules')],
```
Comment (parkjh) [inline android-app/metro.config.js]: @tuncaulubilge are these configs from set template or react native package?

Comment (tuncaulubilge) [reply android-app/metro.config.js]: https://stackoverflow.com/questions/53286974/which-metro-bundler-settings-are-needed-to-resolve-react-natives-internal-modul

**Pull request merged**